src/shell/lisa_gen_buildroot_rootfs

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -eu
+
+BUILDROOT_URI=git://git.busybox.net/buildroot
+BUILDROOT_VERSION=2018.08
+BUILDROOT_DIR=$LISA_HOME/src/buildroot
+BUILDROOT_PATCHES_DIR=$(dirname $0)/buildroot_patches
+BUILDROOT_CONFIG=$(dirname $0)/buildroot_config
+
+function print_usage {
+	echo "Usage: $0 [options]"
+	echo "	options:"
+	echo "		-p: purge buildroot to force a fresh build"
+	echo "		-h: print this help message"
+}
+
+while getopts "hp" opt
+do
+	case $opt in
+		h)
+			print_usage
+			exit 0
+			;;
+		p)
+			rm -rf $BUILDROOT_DIR
+			exit 0
+			;;
+		?)
+			print_usage
+			exit -1
+			;;
+	esac
+done
+
+# Executed command @$1 if file @$2 doesn't exist
+function do_if {
+	if [ ! -e $2 ]; then
```

Comment:
[[ ]] bashism is preferred over simple [ ] since no word splitting or pathname expansion is performed. Note that at the end of the day, double quotes would still be needed because:

> tilde expansion, parameter and variable expansion, arithmetic expansion, command substitution, process substitution, and quote removal are performed.
https://linux.die.net/man/1/bash